provider/user_data.go

```diff
@@ -0,0 +1,72 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/firehydrant/terraform-provider-firehydrant/firehydrant"
+
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceUser() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataFireHydrantPriority,
+		Schema: map[string]*schema.Schema{
+			// Required
+			"email": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			// Computed
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataFireHydrantUser(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	// Get the API client
+	firehydrantAPIClient := m.(firehydrant.Client)
+
+	// Get the user
+	email := d.Get("email").(string)
+	tflog.Debug(ctx, fmt.Sprintf("Fetch user: %s", email), map[string]interface{}{
+		"id": email,
```

Comment:
Update to include the name of what the value is.

```suggestion
		"email": email,
```